### PR TITLE
[Backport release-24.05] discord: format, update discord-canary and discord-ptb, refactor meta

### PR DESCRIPTION
**Pull request by 4evy**
Backports:
https://github.com/NixOS/nixpkgs/pull/347313

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 348303` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>discord-canary</li>
    <li>discord-ptb</li>
  </ul>
</details>

File: pkgs/applications/networking/instant-messengers/discord/linux.nix [hunk 1]
```diff
@@ -1,34 +1,92 @@
...
+  nss,
+  pango,
+  systemd,
+  libappindicator-gtk3,
+  libdbusmenu,
+  writeScript,
+  python3,
+  runCommand,
+  libunity,
+  speechd-minimal,
```
Comment (Artturin) [inline pkgs/applications/networking/instant-messengers/discord/linux.nix:57]: Eval fail from this

**Comment by DontEatOreo** [comment]
From what I see this is the issue:
https://github.com/NixOS/nixpkgs/actions/runs/11315953157/job/31467734934?pr=348303#step:3:56
https://github.com/NixOS/nixpkgs/actions/runs/11315953157/job/31467734934?pr=348303#step:3:74

I think it's because https://github.com/NixOS/nixpkgs/pull/329658 didn't get backported?

Commit adding speechd-minimal: https://github.com/K900/nixpkgs/commit/8e7a0982431b8ad99070695221bdb2d317e08030

**Comment by Artturin** [comment]
Yeah we don't want to backport, change the attr names in this backport

**Comment by DontEatOreo** [comment]
Backported 19afd5603290706849c18ba5ad06684466c45ac9 just before my changes. It seems like it was forgotten to be backported in the last PR

**Comment by DontEatOreo** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348303`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>discord-canary</li>
    <li>discord-ptb</li>
  </ul>
</details>

**Pull request merged**